src/pytorch_lightning/core/module.py

```diff
@@ -291,15 +296,14 @@ def _apply_batch_transfer_handler(
     ) -> Any:
         device = device or self.device
 
-        def call_hook(hook_name, *args):
+        def call_hook(hook_name: str, *args: Any) -> Callable:
             if self._trainer:
                 datahook_selector = self._trainer._data_connector._datahook_selector
                 obj = datahook_selector.get_instance(hook_name)
-                trainer_method = (
-                    self._trainer._call_lightning_module_hook
-                    if isinstance(obj, self.__class__)
-                    else self._trainer._call_lightning_datamodule_hook
-                )
+                if isinstance(obj, self.__class__):
```

Comment:
this is the same logic, it was changed because mypy was complaining 

```sh
src/pytorch_lightning/core/module.py:308: error: Cannot call function of unknown type  [operator]
```